tests/e2e/logging_test.go

```diff
@@ -26,39 +26,60 @@ import (
 )
 
 func TestNoErrorLogsDuringNormalOperations(t *testing.T) {
-	e2e.BeforeTest(t)
-	ctx := context.TODO()
+	tests := []struct {
+		name          string
+		clusterSize   int
+		allowedErrors map[string]bool
+	}{
+		{
+			name:          "single node cluster",
+			clusterSize:   1,
+			allowedErrors: map[string]bool{"setting up serving from embedded etcd failed.": true},
+		},
+		{
+			name:          "three node cluster",
+			clusterSize:   3,
+			allowedErrors: map[string]bool{"setting up serving from embedded etcd failed.": true},
+		},
+	}
 
-	epc, err := e2e.NewEtcdProcessCluster(ctx, t,
-		e2e.WithClusterSize(1),
-		e2e.WithLogLevel("debug"),
-	)
-	require.NoError(t, err)
-	defer epc.Close()
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			e2e.BeforeTest(t)
+			ctx := context.TODO()
 
-	logs := epc.Procs[0].Logs()
-	time.Sleep(time.Second)
-	if err = epc.Close(); err != nil {
-		t.Fatalf("error closing etcd processes (%v)", err)
-	}
-	var entry logEntry
-	lines := logs.Lines()
-	if len(lines) == 0 {
-		t.Errorf("Expected at least one log line")
-	}
+			epc, err := e2e.NewEtcdProcessCluster(ctx, t,
+				e2e.WithClusterSize(tc.clusterSize),
+				e2e.WithLogLevel("debug"),
+			)
+			require.NoError(t, err)
+			defer epc.Close()
+
+			require.Lenf(t, epc.Procs, tc.clusterSize, "embedded etcd cluster process count is not as expected")
+
+			var lines []string
+			for i := range tc.clusterSize {
+				lines = append(lines, epc.Procs[i].Logs().Lines()...)
```

Comment:
Think we need to collect log lines after we close the server.

Reply:
I did encounter the allowed error but apparently not all the time. It indeed does make a difference if `.Lines()` is called after closing. By doing so the resulting log lines seem to be constant. Will update the PR.